src/field_theory/splitting_field.lean

```diff
@@ -109,46 +109,251 @@ namespace splitting_field_aux
+  by exactI Π [distrib_smul α K],
+  by exactI Π [is_scalar_tower α K K] {f : K[X]},
+  α → splitting_field_aux n f → splitting_field_aux n f :=
+nat.rec_on n
+  (λ α K fK ds ist f, by exactI @has_smul.smul _ K _)
+  (λ n ih α K fK ds ist f, by exactI ih α)
+
+instance has_smul (α : Type*) (n : ℕ) {K : Type u} [field K] [distrib_smul α K]
+  [is_scalar_tower α K K] {f : K[X]} :
+  has_smul α (splitting_field_aux n f) :=
+⟨splitting_field_aux.smul n α⟩
 
 instance is_scalar_tower (n : ℕ) : Π (R₁ R₂ : Type*) {K : Type u}
-  [comm_semiring R₁] [comm_semiring R₂] [has_smul R₁ R₂] [field K],
-  by exactI Π [algebra R₁ K] [algebra R₂ K],
-  by exactI Π [is_scalar_tower R₁ R₂ K] {f : K[X]},
-    is_scalar_tower R₁ R₂ (splitting_field_aux n f) :=
-nat.rec_on n (λ R₁ R₂ K _ _ _ _ _ _ _ _, by exactI ‹is_scalar_tower R₁ R₂ K›) $
-         λ n ih R₁ R₂ K _ _ _ _ _ _ _ f, by exactI ih R₁ R₂
+  [has_smul R₁ R₂] [field K],
+  by exactI Π [distrib_smul R₂ K] [distrib_smul R₁ K],
+  by exactI Π [is_scalar_tower R₁ K K] [is_scalar_tower R₂ K K] [is_scalar_tower R₁ R₂ K]
+    {f : K[X]}, by exactI is_scalar_tower R₁ R₂ (splitting_field_aux n f) :=
+nat.rec_on n (λ R₁ R₂ K _ _ hs₂ hs₁ _ _ h f,
+begin
+  rcases hs₁ with @⟨@⟨⟨hs₁⟩,_⟩,_⟩,
+  rcases hs₂ with @⟨@⟨⟨hs₂⟩,_⟩,_⟩,
+  exact h,
+end) $ λ n ih R₁ R₂ K _ _ _ _ _ _ _ f, by exactI ih R₁ R₂
+
+/-- Splitting fields have a negation. -/
+protected def neg (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]},
+  splitting_field_aux n f → splitting_field_aux n f :=
+nat.rec_on n (λ K fK f, by exactI @has_neg.neg K _) (λ n ih K fK f, ih)
+
+/-- Splitting fields have subtraction. -/
+protected def sub (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]},
+  splitting_field_aux n f → splitting_field_aux n f → splitting_field_aux n f :=
+nat.rec_on n (λ K fK f, by exactI @has_sub.sub K _) (λ n ih K fK f, ih)
+
+private lemma nsmul_zero (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]}
+  (x : splitting_field_aux n f), (0 : ℕ) • x = splitting_field_aux.zero n :=
+nat.rec_on n (λ K fK f, by exactI zero_nsmul) (λ n ih K fK f, by exactI ih)
+
+private lemma nsmul_succ (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]}
+  (k : ℕ) (x : splitting_field_aux n f),
+  (k + 1) • x = splitting_field_aux.add n x (k • x) :=
+nat.rec_on n (λ K fK f n x, by exactI succ_nsmul x n) (λ n ih K fK f, by exactI ih)
+
+private lemma zsmul_zero (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]}
+  (x : splitting_field_aux n f), (0 : ℤ) • x = splitting_field_aux.zero n :=
+nat.rec_on n (λ K fK f, by exactI zero_zsmul) (λ n ih K fK f, by exactI ih)
+
+private lemma zsmul_succ (n : ℕ) : Π {K : Type u} [field K], by exactI Π {f : K[X]}
+  (k : ℕ) (x : splitting_field_aux n f),
+  (↑(k + 1) : ℤ) • x = splitting_field_aux.add n x ((k : ℤ) • x) :=
+nat.rec_on n (λ K fK f k x, by exactI @subtraction_comm_monoid.zsmul_succ' K _ k x)
+  (λ n ih K fK f, by exactI ih)
```

Comment:
Is there a reason these aren't handled by the `all_goals` below?

Reply:
I pushed a commit that removes them, it seems to work just fine

Reply:
For some reason the `nsmul` ones don't seem to want to cooperate; I can't figure out why. It's weird because it sometimes works on the UI but not when reloaded/sent to CI.